rust/processor/src/processors/token_v2_processor.rs

```diff
@@ -893,6 +945,22 @@ async fn parse_v2_token(
                             );
                         }
 
+                        // Add burned NFT handling for token datas (can probably be merged with below)
```

Comment:
yeah can we merge this with below since it's doing the same thing

Reply:
the issue is that the code below is more complex than needed here b/c it needs previous owner as well. I'm hoping that as you guys look at sdk you can also rewrite this whole logic. The logic has kind of ballooned out of control a bit. For now I don't see a clean way to merge the paths so just leaving a comment for future.

Reply:
ok i'll take a stab at refactoring this for sdk